packages/nimble-components/src/select/index.ts

```diff
@@ -837,6 +931,41 @@ export class Select
         this.committedSelectedOption = options[this.selectedIndex];
     }
 
+    private toggleNewActiveOption(newActiveIndex: number): void {
+        const selectedIndex = this.openActiveIndex ?? this.selectedIndex;
+        const activeOption = this.options[newActiveIndex];
+        if (newActiveIndex >= 0 && isNimbleListOption(activeOption)) {
+            activeOption.activeOption = true;
+            if (selectedIndex !== newActiveIndex && selectedIndex > -1) {
+                (this.options[selectedIndex]! as ListOption).activeOption = false;
+            }
+
+            if (this.open) {
+                this.openActiveIndex = newActiveIndex;
+                this.focusAndScrollActiveOptionIntoView();
+            } else {
+                this.selectedIndex = newActiveIndex;
+            }
+
+            this.ariaActiveDescendant = this.options[newActiveIndex]?.id ?? '';
```

Comment:
You can use `activeOption` here instead of indexing into `this.options` again, right?